From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Fri, 27 Nov 2020 13:36:21 +0100
Subject: [PATCH] net: phy: micrel: fix interrupt handling

After migration to the shared interrupt support, the KSZ8031 PHY with
enabled interrupt support was not able to notify about link status
change.

Fixes: 59ca4e58b917 ("net: phy: micrel: implement generic .handle_interrupt() callback")
Signed-off-by: Oleksij Rempel <o.rempel@pengutronix.de>
Reviewed-by: Andrew Lunn <andrew@lunn.ch>
Link: https://lore.kernel.org/r/20201127123621.31234-1-o.rempel@pengutronix.de
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
---
 drivers/net/phy/micrel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/micrel.c b/drivers/net/phy/micrel.c
index 0977aee7d396..6332b34a1754 100644
--- a/drivers/net/phy/micrel.c
+++ b/drivers/net/phy/micrel.c
@@ -221,7 +221,7 @@ static irqreturn_t kszphy_handle_interrupt(struct phy_device *phydev)
 		return IRQ_NONE;
 	}
 
-	if ((irq_status & KSZPHY_INTCS_STATUS))
+	if (!(irq_status & KSZPHY_INTCS_STATUS))
 		return IRQ_NONE;
 
 	phy_trigger_machine(phydev);
